### PR TITLE
scan code with cobra

### PR DESCRIPTION
**Pull request by HaoXuan40404**

### CODE REVIEW

File: .ci/ci_check_commit.sh [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    ret=$?
+    if [ $ret -ne 0 ];then
+        LOG_ERROR "FAILED of command: ${command}"
+        exit 1
+    else
+        LOG_INFO "SUCCESS of command: ${command}"
+    fi
+}
+
+init()
```
Comment (bxq2011hust) [inline .ci/ci_check_commit.sh]: this is unnecessary, what about just use HEAD^ below

Comment (HaoXuan40404) [reply .ci/ci_check_commit.sh]: It happened when repo only has first commit, it's better than nothing

File: .travis.yml [hunk 2]
```diff
@@ -6,36 +6,36 @@ branches:
...
-  ./gradlew test
-  ./gradlew integrationTest
+    - language: java
+      jdk: openjdk8
+      os: linux
+      dist: xenial
+      sudo: required
+      before_cache:
+        - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
+        - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
```
Comment (bxq2011hust) [inline .travis.yml]: just add a new python build, without modifying the config before

Comment (HaoXuan40404) [reply .travis.yml]: add more ci now

**Pull request merged**